pseudo.go

```diff
@@ -236,12 +293,53 @@ func (n *node) pushExcess() {
 }
 
 // (*node) breakRelationship
+//static inline void
+func (n *node) breakRelationship(child *node) {
+	var current *node
+	oldParent := n
+	child.parent = nil
+
+	if oldParent.childList == child {
+		oldParent.childList = child.next
+		child.next = nil
+		return
+	}
+
+	for current = oldParent.childList; current.next != child; current = current.next { //TODO: check
+		current.next = child.next
+		child.next = nil
+	}
+}
 
 // (*node) addRelationship
+// static inline int
+func (n *node) addRelationship(child *node) int {
+	newParent := n
+	child.parent = newParent
+	child.next = newParent.childList
+	newParent.childList = child
+
+	return 0
+}
 
 // (*node) findWeakNode(weakNode *node)
 
 // (*node) checkChildren
+func (n *node) checkChildren(curNode *node) {
+	//FIXME
+	for ; curNode.nextScan; curNode.nextScan = curNode.nextScan.next {
```

Comment:
should be
`for ; curNode.nextScan != nil; curNode.nextScan = curNode.nextScan.next {`